src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/SynchronizedMethodHelpers.cs

```diff
@@ -25,13 +25,7 @@ private static void MonitorEnter(object obj, ref bool lockTaken)
                 ObjectHeader.GetLockObject(obj) :
                 SyncTable.GetLockObject(resultOrIndex);
 
-            if (lck.TryAcquire(0))
```

Comment:
This acquire attempt is unnecessary and is also too soon. It was probably here to pick "unusual" cases like recursive locking by deferring internally to the slow path for one iteration. Calling the slow path is simpler and just as good.